drivers/scheduler/k8s/k8s.go

```diff
@@ -6951,7 +6951,29 @@ func (k *K8s) snapshotAndVerify(size resource.Quantity, data, snapName, namespac
 			return fmt.Errorf("restored volume does NOT contain data from original volume: expected to contain '%s', got '%s'", data, string(fileContent))
 		}
 	}
-	log.Info("Validation complete")
+
+	log.Info("Validation complete, deleting restored pods")
+	err = clientset.CoreV1().Pods(namespace).Delete(context.TODO(), restoredPod.Name, metav1.DeleteOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return fmt.Errorf("error deleting restored pod: %s", err)
+	}
+
+	// Wait for the pod to be actually gone
+	t := func() (interface{}, bool, error) {
+		var err error
+		if _, err = clientset.CoreV1().Pods(namespace).Get(context.TODO(), restoredPod.Name, metav1.GetOptions{}); err != nil {
+			if k8serrors.IsNotFound(err) {
+				return nil, false, nil // Pod is now successfully gone
+			}
+			return nil, true, fmt.Errorf("failed to check that pod %s in ns %s is gone. Err: %v", restoredPod.Name, restoredPod.Namespace, err)
+		}
+		return nil, true, fmt.Errorf("pod %s in ns %s still exists", restoredPod.Name, restoredPod.Namespace)
+	}
+
+	if _, err := task.DoRetryWithTimeout(t, time.Minute*2, DefaultRetryInterval); err != nil {
+		return err
+	}
```

Comment:
I think this code is a bit "racey" -- it may happen that the K8s is very quick with restarting the PODs after they got deleted -- and we'd get "still exists" error.
I think a better approach would be to collect the POD-IDs before the deletion, then inside `t()` check if those specific POD-IDs are now gone.

Reply:
@zoxpx in this case the pods are not managed by a deployment or anything (they are one-off pods), so they won't be restarted

Reply:
Well.. first off, do we actually need to wait until the pods are gone?
* note that the `kubectl delete pod -Al name=portworx` will actually hang until all portworx PODs will be in `Terminated` state
* not sure if K8s API behaves the same way -- but if it does, probably no need to have the extra code that waits for the PODs to actually disappear

Other than that, I don't think there's a big danger that PODs get restarted w/ the exact same UID --
* if the POD executable dies/crashes and POD gets restarted, then yes
* but I haven't actually seen the PODs getting restarted w/ the exact same name + UID after the successful 'pod delete' function..  (they _will_ recycle POD name/UID once restarted).

Finally, my apologies, I haven't initially noticed that you are dealing with a single POD, and you're deleting it "by name" (I thought you were deleing a group of PODs).   My initial feedback was based on this assumption -- but I think I was wrong, and I think this code actually works OK.

To speed things up, you might want to also check for the status of the POD -- and break out immediately when the status switches to `Terminated`  (AFAIK Terminated PODs are like zombie-processes, just need to be cleaned up from process-table, which could take a minute on a busy K8s cluster)

Reply:
We actually specifically want to wait for pods to be fully terminated: we've had a lot of bugs where FlashArray/FlashBlade pods get stuck in terminating, and we want this to fail loudly on it.